.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,160 @@
+
+*First of all, thank you for taking the time to contribute!*
+
+The following is a set of guidelines for contributing to ZFS on Linux. 
+Using them you can help us to make a better work.
+
+#### Table Of Contents
+
+[What should I know before I get started?](#what-should-i-know-before-i-get-started)
+
+  * [Get ZFS](#get-zfs)
+  * [Debugging ZFS](#debugging-zfs)
+  * [Where can I ask for help?](#where-can-I-ask-for-help)
+  
+[How Can I Contribute?](#how-can-i-contribute)
+
+  * [Reporting Bugs](#reporting-bugs)
+  * [Suggesting Enhancements](#suggesting-enhancements)
+  * [Pull Requests](#pull-requests)
+  * [Testing](#testing)
+  
+[Styleguides](#styleguides)
+ 
+  * [Coding conventions](#coding-conventions)
+
+Helpful resources
+
+  * [ZFS on Linux wiki](https://github.com/zfsonlinux/zfs/wiki)
+  * [OpenZFS Documentation](http://open-zfs.org/wiki/Developer_resources)
+
+
+## What should I know before I get started?
+
+### Get ZFS
+
+You can build fresh zfs packages using 
+[this instruction](https://github.com/zfsonlinux/zfs/wiki/Building-ZFS), 
+or use stable packages from 
+[your distribution repository](https://github.com/zfsonlinux/zfs/wiki/Getting-Started).
+
+### Debugging ZFS
+
+`zdb` utility is the main tool to debug ZFS. 
+
+Additionally, see 
+[build instruction](https://github.com/zfsonlinux/zfs/wiki/Building-ZFS)
+, there is `--enable-debug` flag, which can be set to enable all ASSERTs and additional correctness tests.
+
+### Where can I ask for help?
+The best place to ask is our [mailing list](https://github.com/zfsonlinux/zfs/wiki/Mailing-Lists).
+
+Please *note*, our issue tracker is **only for bugs**, if you have thoughts about any 
+enhancements - first tell about them via 
+[mailing list](https://github.com/zfsonlinux/zfs/wiki/Mailing-Lists).
+
+## How Can I Contribute?
+
+### Reporting Bugs
+
+You can open a new issue and search existing issues using the public
+[issue tracker](https://github.com/zfsonlinux/zfs/issues). 
+
+Please make sure that what you're actually seeing is a bug and **not a support issue**. 
+
+If in doubt, please ask on the [mailing list](https://github.com/zfsonlinux/zfs/wiki/Mailing-Lists) 
+first, and if you're then asked to file an issue, do so.
+
+####When opening a new issue include this information at the top of the issue:
+
+* What distribution you're using and the version.
+* What spl/zfs packages you're using and the version.
+* Describe the problem you're observing.
+* Describe how to reproduce the problem.
+* Including any warning/errors/backtraces from the system logs.
+
+When a new issue is opened it's not uncommon for a developer to request additional 
+information about the problem. 
+
+In general, the more detail you share about a problem the quicker a developer 
+can resolve it. For example, providing a simple test case is always 
+exceptionally helpful.  
+
+Be prepared to work with the developer looking in to your bug in order to get 
```

Comment:
`Be prepared to work with the developer investigating your issue. Your assistance is crucial in providing a quick solution.` Thoughts?